src/App.js

```diff
@@ -1,24 +1,43 @@
-import logo from './logo.svg';
-import './App.css';
+import React, {useState, useEffect} from 'react';
+import Course from './screens/course/Course';
+import Lesson from './screens/lesson/Lesson';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { supabase } from './supabaseClient';
 
 function App() {
+  const [courseItems, setCourseItems] = useState([]);
+
+  const getCourseItems = async () => {
+    const { data, error } = await supabase
+      .from('courses')
+      .select('name, id')
+
+    if(error) {
+      throw new Error(error.message)
+    }
+    if(!data) {
+      throw new Error("Course not found")
+    }
+    setCourseItems(data);
+  };
+
+  useEffect(() => {
+    getCourseItems();
+  });
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Course />} />
+        {courseItems.map((course, index) => (
+          <Route 
+            key={index} 
+            path={`/${course.name}/lesson/`} 
```

Comment:
Please use slug in URL. For example course name is `The master class`, the slug will be `the-master-class`

Reply:
Will change

Reply:
Edited
https://github.com/cam-drei/kids-study-front-end/pull/1/commits/9481027a082b99cf1b7fd21f2c4b2f97a623d484